datajunction-server/datajunction_server/construction/build.py

```diff
@@ -796,12 +809,27 @@ def build_source_node_query(node: NodeRevision):
 def build_ast(  # pylint: disable=too-many-arguments
     session: Session,
     query: ast.Query,
+    memoized_queries: Dict[int, ast.Query] = None,
     build_criteria: Optional[BuildCriteria] = None,
 ) -> ast.Query:
     """
     Determines the optimal way to build the query AST and does so
     """
+    if not memoized_queries:
+        memoized_queries = {}
```

Comment:
This is equivalent to setting the default in the function signature--`memoized_queries: Dict[int, ast.Query] = {}`

Reply:
Actually I think doing that will cause other side effects: https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python

Reply:
Woah, I learned something new. 🧐

Reply:
References!

Reply:
Do people generally like this instead of `memoized_queries = memoized_queries or {}`? I've also wondered why this common pattern hasn't risen to the level of the case that inspired walrus notation in python `:=`

Reply:
Ah yeah, that one-liner is cleaner! Let me update